File: back end/ordening-service/src/main/java/com/cibertec/service/OrderService.java
package com.cibertec.service;

import java.util.List;

import com.cibertec.dto.OrderRequest;
import com.cibertec.dto.OrderResponse;

public interface OrderService {
    
    List<OrderResponse> getAllOrders();

    OrderResponse getOrderById(Long id);

    OrderResponse createOrder(OrderRequest orderRequest);

    OrderResponse updateOrder(Long id, OrderRequest orderRequest);
    
    void deleteOrder(Long id);
}

File: la-gula-pizzeria/src/app/shared/services/payment/order-service.ts
import { Injectable } from '@angular/core';
import { GATEWAY_URL, SERVICES_PATHS } from '../config/config';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import OrderResponse from '../../model/payment/response/orderResponse.model';


@Injectable({
  providedIn: 'root'
})
export class OrderService {
  path = `${GATEWAY_URL}${SERVICES_PATHS.PAYMENT}` + '/orders';

  constructor(private http: HttpClient) { }

  getAllOrders() : Observable<OrderResponse[]> {
    return this.http.get<OrderResponse[]>(this.path);
  }
  getOrderById(id: number) : Observable<OrderResponse> {
    return this.http.get<OrderResponse>(`${this.path}/${id}`);
  }
  saveOrder(order: OrderResponse) : Observable<OrderResponse> {
    return this.http.post<OrderResponse>(this.path, order);
  }
  updateOrder(id: number, order: OrderResponse) : Observable<OrderResponse> {
    return this.http.put<OrderResponse>(`${this.path}/${id}`, order);
  }
  deleteOrder(id: number) : Observable<void> {
    return this.http.delete<void>(`${this.path}/${id}`);
  }
}